unbind: free socket prior to destroying TLS context

Author: Jan Vcelak <jvcelak@redhat.com>
Resolves: #808464
Upstream ITS: #7241
Upstream commit: 221531b31a02c162c60c7d4491169855454c1ee0

---
 libraries/libldap/unbind.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libraries/libldap/unbind.c b/libraries/libldap/unbind.c
index 35a92a8..82bc12d 100644
--- a/libraries/libldap/unbind.c
+++ b/libraries/libldap/unbind.c
@@ -129,6 +129,9 @@ ldap_ld_free(
 		ld->ld_abandoned = NULL;
 	}
 	LDAP_MUTEX_UNLOCK( &ld->ld_res_mutex );
+
+	ber_sockbuf_free( ld->ld_sb );
+
 	LDAP_MUTEX_LOCK( &ld->ld_ldopts_mutex );
 
 	/* final close callbacks */
@@ -217,8 +220,6 @@ ldap_ld_free(
 	}
 	LDAP_MUTEX_UNLOCK( &ld->ld_ldopts_mutex );
 
-	ber_sockbuf_free( ld->ld_sb );   
-   
 #ifdef LDAP_R_COMPILE
 	ldap_pvt_thread_mutex_destroy( &ld->ld_msgid_mutex );
 	ldap_pvt_thread_mutex_destroy( &ld->ld_conn_mutex );
-- 
1.7.10.4

